---
# Display name
title: James A. Fellows Yates

# Is this the primary user of the site?
superuser: true

# Role/position/tagline
role: Bioinformatician and Biomolecular Archaeologist

# Organizations/Affiliations to show in About widget
organizations:
- name: Leibniz-Institut für Naturstoff-Forschung und Infektionsbiologie Hans-Knöll-Institut
  url: https://www.leibniz-hki.de/en/

# Short bio (displayed in user profile at end of posts)
bio: My research interests focus on biomolecular archaeology, ancient DNA, metagenomics and bioinformatics

# Interests to show in About widget
interests:
- Ancient Metagenomics
- Bioinformatics

# Education to show in About widget
education:
  courses:
  - course: Ph.D. in Bioinformatics
    institution: FSU Jena / MPI-SHH
    year: 2021
  - course: M.Sc. in Naturwissenschaftliche Archäologie
    institution: Eberhard Karls Universität Tübingen
    year: 2015
  - course: BSc in Bioarchaeology
    institution: University of York
    year: 2013

# Social/Academic Networking
# For available icons, see: https://wowchemy.com/docs/getting-started/page-builder/#icons
#   For an email link, use "fas" icon pack, "envelope" icon, and a link in the
#   form "mailto:your-email@example.com" or "/#contact" for contact widget.
social:
- icon: envelope
  icon_pack: fas
  link: '/#contact'
- icon: twitter
  icon_pack: fab
  link: https://twitter.com/jfy133
- icon: github
  icon_pack: fab
  link: https://github.com/jfy133
- icon: google-scholar  # Alternatively, use `google-scholar` icon from `ai` icon pack
  icon_pack: ai
  link: https://scholar.google.com/citations?user=5Wy_JDgAAAAJ&hl=en
- icon: orcid
  icon_pack: ai
  link: https://orcid.org/0000-0001-5585-6277
- icon: linkedin
  icon_pack: fab
  link: https://www.linkedin.com/in/james-fellows-yates-999859179/
- icon: cv
  icon_pack: ai
  link: https://registry.jsonresume.org/jfy133

# Link to a PDF of your resume/CV.
# To use: copy your resume to `static/uploads/resume.pdf`, enable `ai` icons in `params.toml`, 
# and uncomment the lines below.


# Enter email to display Gravatar (if Gravatar enabled in Config)
email: "jfy133@gmail.com"

# Highlight the author in author lists? (true/false)
highlight_name: false
---
b
I am a biomolecular archaeologist and bioinformatician. I currently work between the Department of [Palaeobiotechnology](https://www.leibniz-hki.de/en/paleobiotechnology.html) at the Leibniz Institute for Natural Product Research and Infection Biology Hans Knöll Institute and [Max Planck Institute for Evolutionary Anthropology](https://www.eva.mpg.de), while finishing up my [my Ph.D.](http://www.shh.mpg.de/employees/45083/25522) on [ancient oral microbiomes](http://www.shh.mpg.de/348251/Evolution-and-Ecology-of-the-Human-Oral-Microbiome) .

I originally trained as an Archaeologist at the University of York, where I specialised in palaeoproteomics. After which I moved to the Universität Tübingen to study for an M.Sc. in Archaeological Sciences, specialising in palaeogenetics. During my time in Tübingen I also worked as a technician for a stable isotope lab. I worked for a short period for the [Institut für Vor- und Frühgeschichtliche Archäologie and Provinzialrömische Archäologie](https://www.vfp-archaeologie.uni-muenchen.de/ueber_uns/index.html) at LMU München, developing strategies for authenticating ancient dietary DNA.

My current area of research is centred on the ancient DNA content of fossilised dental plaque, also known as dental calculus. My sample set includes individuals spanning from the Palaeolithic to modern day. I am primarily exploring how we can analyse and [authenticate](http://www.jafy.eu/publication/2017-04-17-robust-framework-microbial-archaeology) the complex genetic make-up of this material through shotgun metagenomics.

I am involved in the leading and developing of the open-source nexflow pipeline [nf-core/eager (v2)](https://github.com/nf-core/eager), which is intended to be a complete re-write and extension of the NGS processing pipeline for ancient DNA data [EAGER (v1)](https://doi.org/10.1186/s13059-016-0918-z), to reach modern computational and palaeogenetic standards. I am developing and contributing to multiple open-source software focusing on ancient metagenomics (which can be seen on my [github](https://www.github.com/jfy133/)). 

I also established an [international community of researchers in ancient metagenomics](https://spaam-community.github.io), and ran the associated [SPAAM2 round-table workshop](https://spaam-community.github.io/#/spaam2/README), and I am also a core-team member of the [nf-core](https://nf-co.re) initiative for best-practise Nextflow bioinformatic pipelines.

Previously, I have worked on [Late Pleistocene woolly mammoth mitochondrial genomes](http://www.jafy.eu/publication/2017-12-18-central-european-woolly-mammoth), as well as testing [collagen 'fingerprinting' methods (ZooMS) on burnt skeletal remains](https://www.researchgate.net/publication/274006074_It_Will_Not_Be_Possible_To_Use_Zooarchaeology_By_Mass_Spectrometry_ZooMS_To_Identify_Species_In_Samples_Of_Cremated_Bone_That_Have_Been_Burnt_Higher_Than_155oC?_iepl%5BviewId%5D=hOoc4o8VsWpiJjo3eXvIspCV&_iepl%5BprofilePublicationItemVariant%5D=default&_iepl%5Bcontexts%5D%5B0%5D=prfpi&_iepl%5BtargetEntityId%5D=PB%3A274006074&_iepl%5BinteractionType%5D=publicationTitle).

{{< icon name="download" pack="fas" >}} See my <a href="https://registry.jsonresume.org/jfy133">resumé</a>.
